server/src/actors/project.rs

```diff
@@ -89,13 +96,29 @@ impl Project {
     ) -> Response<Arc<ProjectState>, ProjectError> {
         // count number of times we are looking for the project state
         metric!(counter("project_state.get") += 1);
-        if let Some(ref state) = self.state {
-            // In case the state is fetched from a local file, don't use own caching logic. Rely on
+
+        let state = self
+            .state
+            .as_ref()
+            .map(|s| (s, s.outdated(self.id, &self.config)));
```

Comment:
You could simplify this and the match below by defaulting outdated to `Outdated::HardOutdated`.